axelrod/result_set.py

```diff
@@ -14,25 +16,30 @@
 class ResultSet(object):
     """A class to hold the results of a tournament."""
 
-    def __init__(self, players, matches, with_morality=True):
+    def __init__(self, players, interactions, with_morality=True):
         """
         Parameters
         ----------
             players : list
                 a list of player objects.
-            matches : list
+            interactions : list
                 a list of dictionaries mapping tuples of player indices to
-                completed matches (1 for each repetition)
+                interactions (1 for each repetition)
             with_morality : bool
                 a flag to determine whether morality metrics should be
                 calculated.
         """
         self.players = players
         self.nplayers = len(players)
-        self.matches = matches
-        self.nrepetitions = len(matches)
+        self.interactions = interactions
+        self.nrepetitions = len(interactions)
 
         # Calculate all attributes:
+        self.build_all(with_morality)
+
+    def build_all(self, with_morality):
```

Comment:
Perhaps something to investigate at a later stage perhaps?

I suggest that users in a notebook etc will be running small enough tournaments that it makes sense to just calculate everything (the order in which this sets the attribute is purposeful).

If however you're running a big tournament then you would probably want to do a one of calculation... Or perhaps that's not the right way to think about it...

>  If anything is particularly intense or less likely to be used then they would be good candidates for properties.

I would prefer it if we handled every property the same I think...